query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala

```diff
@@ -613,4 +620,27 @@ object QueryRewriter {
       case None => fc
     }
   }
+
+  private def topLevelAndContain(target: Expr, expr: Expr): Boolean = {
+    expr match {
+      case e: Expr if e == target =>
+        true
+      case fc: FunctionCall if fc.function.name == And.name =>
+        topLevelAndContain(target, fc.parameters.head) ||
+          topLevelAndContain(target, fc.parameters.tail.head)
+      case _ =>
+        false
+    }
+  }
+
+  private def stripExprInTopLevelAnd(target: Expr, expr: Expr): Expr = {
```

Comment:
I thought about combining the two functions together.  But I expect that will make it significantly harder to reason about at best.